homeassistant/components/tod/binary_sensor.py

```diff
@@ -42,6 +48,42 @@
 )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Times of the Day config entry."""
+    if hass.config.time_zone is None:
+        _LOGGER.error("Timezone is not set in Home Assistant configuration")
+        return
```

Comment:
This should move to the integration entry setup, so we can fail the setup state (or make it retry).